joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -114,12 +115,24 @@ JointTrajectoryController::state_interface_configuration() const
       conf.names.push_back(joint_name + "/" + interface_type);
     }
   }
+  conf.names.push_back(params_.speed_scaling_interface_name);
   return conf;
 }
 
 controller_interface::return_type JointTrajectoryController::update(
   const rclcpp::Time & time, const rclcpp::Duration & period)
 {
+  if (state_interfaces_.back().get_name() == params_.speed_scaling_interface_name)
+  {
+    scaling_factor_ = state_interfaces_.back().get_value();
+  }
+  else
+  {
+    RCLCPP_ERROR(
```

Comment:
Using a throttled logger or
```suggestion
    RCLCPP_ERROR_ONCE(
```

Reply:
I'll leave this unresolved for now. Once the scaling interface is optional it would probably make sense to return ERROR at this point (If scaling is configured to be used, but no interface is found).